### ISSUE TITLE
Unhandled Exception: System.ArgumentOutOfRangeException: Table ' was not found

### ISSUE THREAD

**Issue by augustoproiete** [ClosedXML/ClosedXML#1683]
**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

**Version of ClosedXML**

0.95.4

**What is the current behavior?**

Opening files that have external links to other files open fine in Excel, but fail to open with ClosedXML with the error `System.ArgumentOutOfRangeException: Table ' was not found`. Reproducible example: [test.xlsx](https://github.com/ClosedXML/ClosedXML/files/6705009/test.xlsx)

```
Unhandled Exception: System.ArgumentOutOfRangeException: Table ' was not found.
Parameter name: name
   at ClosedXML.Excel.XLTables.Table(String name) in C:\ClosedXML\Excel\Tables\XLTables.cs:line 79
   at ClosedXML.Excel.XLWorksheet.Table(String name) in C:\ClosedXML\Excel\XLWorksheet.cs:line 615
   at ClosedXML.Excel.XLWorksheet.Range(String rangeAddressStr) in C:\ClosedXML\Excel\XLWorksheet.cs:line 1663
   at ClosedXML.Excel.XLWorksheet.ClosedXML.Excel.IXLWorksheet.Range(String rangeAddress) in C:\ClosedXML\Excel\XLWorksheet.cs:line 507
   at ClosedXML.Excel.XLSparklineGroup.Add(String locationRangeAddress, String sourceDataAddress) in C:\ClosedXML\Excel\Sparkline\XLSparkLineGroup.cs:line 151
   at ClosedXML.Excel.XLWorkbook.<>c__DisplayClass314_1.<LoadExtensions>b__4(Sparkline sl) in C:\ClosedXML\Excel\XLWorkbook_Load.cs:line 2740
   at ClosedXML.Excel.EnumerableExtensions.ForEach[T](IEnumerable`1 source, Action`1 action) in C:\ClosedXML\Extensions\EnumerableExtensions.cs:line 12
   at ClosedXML.Excel.XLWorkbook.LoadExtensions(WorksheetExtensionList extensions, XLWorksheet ws) in C:\ClosedXML\Excel\XLWorkbook_Load.cs:line 2739
   at ClosedXML.Excel.XLWorkbook.LoadSpreadsheetDocument(SpreadsheetDocument dSpreadsheet) in C:\ClosedXML\Excel\XLWorkbook_Load.cs:line 303
   at ClosedXML.Excel.XLWorkbook.LoadSheets(Stream stream) in C:\ClosedXML\Excel\XLWorkbook_Load.cs:line 49
   at ClosedXML.Excel.XLWorkbook.Load(Stream stream) in C:\ClosedXML\Excel\XLWorkbook_Load.cs:line 37
   at ClosedXML.Excel.XLWorkbook..ctor(Stream stream, XLEventTracking eventTracking) in C:\ClosedXML\Excel\XLWorkbook.cs:line 739
   at ClosedXML_Sandbox.Program.Main(String[] args) in C:\ClosedXML_Sandbox\Program.cs:line 15
```

**What is the expected behavior or new feature?**

Excel file should open successfully.

**Is this a regression from the previous version?**

**Yes**. The same [repro](https://github.com/ClosedXML/ClosedXML/files/6705009/test.xlsx) file works fine with ClosedXML v0.94.2. It seems the issue was introduced with v0.95.0

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 

## Reproducibility

```c#
using ClosedXML.Excel;

public void Main()
{
    var _ = new XLWorkbook(@"C:\augustoproiete\test.xlsx");
}
```

- [x] I attached a sample spreadsheet: **[test.xlsx](https://github.com/ClosedXML/ClosedXML/files/6705009/test.xlsx)**

**Comment by TheByrd92** [comment]
I'm not sure if this is your same issue because your excel sheet doesn't seem to have tables in the example but, this error is happening from...

```
public IXLTable Table(String name)
{
   if (TryGetTable(name, out XLTable table))
      return table;
   throw new ArgumentOutOfRangeException(nameof(name), $"Table {name} was not found.");
}

internal bool TryGetTable(string tableName, out XLTable table)
{
   return _tables.TryGetValue(tableName, out table);
}
```
In the XLTables.cs

Called from...
```
var ws = WorksheetsInternal.FirstOrDefault<XLWorksheet>(w => w.SheetId == (localSheetId + 1));
if (ws != null)
   {
      ws.PageSetup.PrintAreas.Add(area);
   }
}
```
When it tries to find a table instead of a range at this point ```TryGetTable``` it does have the table in the worksheet. But, it's grabbing the wrong sheetid from the definednames or at least an unexpected sheetid from defined names when setting up the print areas. It seems to have something to do with ranges as tables. ```Table7[#All]``` like that. It's being assigned the incorrect or unexpected sheetId when the openxml.workbook.definedNames is iterated through under certain circumstances that I don't know where.

TL:DR
Long story short localSheetId may be an unreliable number to use for gathering range information for print areas.
